src/services/types.ts

```diff
@@ -403,7 +403,12 @@ namespace ts {
     // See `commands?: {}[]` in protocol.ts
     export type CodeActionCommand = InstallPackageAction;
 
-    export interface InstallPackageAction {
+    /* @internal */
+    export interface CodeActionCommandBase {
+        file: string;

```

Comment:
Might the project name be required as well if the same file appears in multiple projects?

Reply:
It seems like it will be required if we are going to call `updateTypingsForProject` (which calls `delayUpdateProjectGraphAndInferredProjectsRefresh`).

Reply:
This is an opaque object so we can change this if we turn out to need that.
I think we should update *all* projects that include that file, since they all might be affected by a new directory in `node_modules`. They should already be watching that directory though.

Reply:
I would have thought `PackageInstalledResponse` would behave more or less the same as `SetTypings`, which is a `ProjectResponse`.